src/Altinn.App.Core/Implementation/DefaultTaskEvents.cs

```diff
@@ -256,6 +268,44 @@ private async Task RemoveHiddenData(Instance instance, Guid instanceGuid, List<D
         }
     }
 
+    private async Task RemoveShadowFields(Instance instance, Guid instanceGuid, List<DataType> dataTypesToLock)
+    {
+        foreach (var dataType in dataTypesToLock.Where(dt => dt.AppLogic?.ShadowFields != null))
+        {
+            foreach (Guid dataElementId in instance.Data.Where(de => de.DataType == dataType.Id).Select(dataElement => Guid.Parse(dataElement.Id)))
+            {
+                Type modelType = _appModel.GetModelType(dataType.AppLogic.ClassRef);
+                string app = instance.AppId.Split("/")[1];
+                int instanceOwnerPartyId = int.Parse(instance.InstanceOwner.PartyId);
+                dynamic data = await _dataClient.GetFormData(
+                    instanceGuid, modelType, instance.Org, app, instanceOwnerPartyId, dataElementId);
+
+                var modifier = new IgnorePropertiesWithPrefix(dataType.AppLogic.ShadowFields.Prefix);
+                JsonSerializerOptions options = new ()
+                {
+                    TypeInfoResolver = new DefaultJsonTypeInfoResolver
+                    {
+                        Modifiers = { modifier.ModifyPrefixInfo }
+                    },
+                    DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+                };
+                
+                string serializedData = JsonSerializer.Serialize(data, options);
+                if (dataType.AppLogic.ShadowFields.SaveToDataType != null) {
+                    var saveToDataType = dataTypesToLock.Find(dt => dt.Id == dataType.AppLogic.ShadowFields.SaveToDataType);
+                    Type saveToModelType = _appModel.GetModelType(saveToDataType.AppLogic.ClassRef);
+                    var updatedData = Newtonsoft.Json.JsonConvert.DeserializeObject(serializedData, saveToModelType ?? modelType);
```

Comment:
Is there a reason for using Newtonsoft instead of System.Text.Json here?

Reply:
Only that it is consistent with our deserialization elsewhere in this project (as far as I could tell). Had to use `System.Text.Json` for the actual removal of fields, as Newtonsoft did not seem to support that kind of operation.

Reply:
We are migrating away from Newtonsoft, but haven't gotten around to removing it from the entire project. New code is, unless there are good reasons, leveraging System.Text.Json and when we touch code using Newtonsoft we migrate it to System.Text.Json.